app/websocket/vault_socket.py

```
@@ -154,25 +154,92 @@ async def websocket_user_endpoint(
+    Firmware sends: heartbeat, state_update, ack
+    Backend sends: remote_unlock, buzzer_off, reset
     """
-    
-    raise WebSocketException(
-        code=status.WS_1008_POLICY_VIOLATION,
-        reason="Vault device websocket not wired yet",
+    from app.api.deps.common import get_db_session
+    from app.infrastructure.db.repositories.sqlalchemy_activity_log_repository import (
+        SqlAlchemyActivityLogRepository,
     )
+    from app.application.use_cases.process_vault_state_update import ProcessVaultStateUpdate
```

Ruff (0.15.2)

[error] Redefinition of unused `ProcessVaultStateUpdate` from line 18: `ProcessVaultStateUpdate` redefined here

Remove definition: `ProcessVaultStateUpdate`

(F811)